kv/kvserver: TestLeaseRenewer failed

kv/kvserver.TestLeaseRenewer [failed](https://tanzanite.cluster.engflow.com/invocations/default/e64b96a6-7039-4853-8fbb-c6e9db4782ae?testReportRun=3&testReportShard=24&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on release-24.1.0-rc @ [5e4ca9e26f1a25681de9c944298cfa139c344466](https://github.com/cockroachdb/cockroach/commits/5e4ca9e26f1a25681de9c944298cfa139c344466):


```
=== RUN   TestLeaseRenewer
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestLeaseRenewer3541764033
    test_log_scope.go:81: use -show-logs to present logs inline
    replica_lease_renewal_test.go:144: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseRenewer3541764033
--- FAIL: TestLeaseRenewer (85.52s)
=== RUN   TestLeaseRenewer/kv.expiration_leases_only.enabled=true
    replica_lease_renewal_test.go:103: r1 lease extended: repl=(n2,s2):2 seq=2 start=1717496177.389160515,0 exp=1717496193.122723394,0 pro=1717496187.122723394,0
    replica_lease_renewal_test.go:103: r1 lease extended: repl=(n2,s2):2 seq=2 start=1717496177.389160515,0 exp=1717496193.473117872,0 pro=1717496187.473117872,0
    replica_lease_renewal_test.go:103: r1 lease extended: repl=(n2,s2):2 seq=2 start=1717496177.389160515,0 exp=1717496193.872532082,0 pro=1717496187.872532082,0
    replica_lease_renewal_test.go:85: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_lease_renewal_test.go:85
        	            				github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_lease_renewal_test.go:92
        	            				github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_lease_renewal_test.go:125
        	            				github.com/cockroachdb/cockroach/pkg/testutils/subtest.go:28
        	Error:      	Received unexpected error:
        	            	r69 was not found on s1
        	Test:       	TestLeaseRenewer/kv.expiration_leases_only.enabled=true
    --- FAIL: TestLeaseRenewer/kv.expiration_leases_only.enabled=true (48.86s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=24</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseRenewer.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39210

--- a/pkg/kv/kvserver/replica_lease_renewal_test.go
+++ b/pkg/kv/kvserver/replica_lease_renewal_test.go
@@ -122,11 +122,13 @@ func TestLeaseRenewer(t *testing.T) {
 
 		// Split off an expiration-based range, and assert that the lease is extended.
 		desc := tc.LookupRangeOrFatal(t, tc.ScratchRangeWithExpirationLease(t))
+		// Wait for the split to apply on all nodes first.
+		require.NoError(t, tc.WaitForFullReplication())
+		// Assert that the lease is extended.
 		assertLeaseExtension(desc.RangeID)
 
 		// Transfer the lease to a different leaseholder, and assert that the lease is
-		// still extended. Wait for the split to apply on all nodes first.
-		require.NoError(t, tc.WaitForFullReplication())
+		// still extended.
 		lease, _ := getNodeReplica(1, desc.RangeID).GetLease()
 		target := tc.Target(lookupNode(lease.Replica.NodeID%3 + 1))
 		tc.TransferRangeLeaseOrFatal(t, desc, target)

